### DISCUSSION TITLE
"Artifact update problem" for Helm OCI registries when username contains the `$` character

### DISCUSSION THREAD

**Discussion by dominykas** [renovatebot/renovate#26373] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.92.4

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

1. We user Harbor registry with robot accounts - the robot account name is generated and contains the `$` sign, e.g. `robot$library-dev+jenkins-dev`
2. We have `hostRules` set up for the Harbor domain
3. We have a Helm chart which has another Helm chart as a dependency, coming from the above Harbor registry

When there's a new version, we get the following "Artifact update problem" failure in the PR:

```
Command failed: helm registry login --username 'robot$ci-jenkins-prd' --password **redacted** harbor.example.com/library
WARNING: Using --password via the CLI is insecure. Use --password-stdin.
Error: Get "https://harbor.example.com/v2/": unauthorized: authentication required
```

I can work around this if I stick in a `\` before the `$` in the configuration, however that would probably break other use cases (e.g. API calls when updating actual Docker images).

I find it odd, that running the `helm registry login` command with user name quoted in single quotes locally does actually work. It also works for an unquoted username where `$` is escaped (i.e. `\$`). And of course it does not work when `$` is used as is. Is there something that `child_process.spawn` does to the quotes in the args?

I do not have a public way to test this, given that it requires specific usernames...

### Relevant debug logs

<details><summary>Logs</summary>
```

```
</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
We "quote" CLI arguments, that could potentially be causing it to get corrupted

**Comment by dominykas** [comment]
Yeah, that's the interesting thing - if I copy/paste the command as printed in the logs (with all the quotes) - it seems to work OK.
